sorting/quick_sort_3.cpp

```diff
@@ -0,0 +1,188 @@
+/**
+ * @file
+ * @brief Implementation Details
+ * @details Quick sort 3 works on Dutch National Flag Algorithm
+ * The major difference between simple quicksort and quick sort 3 comes in the
+ * function partition3 In quick_sort_partition3 we divide the vector/array into
+ * 3 parts. quick sort 3 works faster in some cases as compared to simple
+ * quicksort.
+ * @author immortal-j
```

Comment:
```suggestion
 * @author [immortal-j](https://github.com/immortal-j)
```

Reply:
this was authors' contribution. will leave it as is.

Reply:
Still not resolved...